modules/core/src/test/java/org/apache/ignite/internal/processors/security/rest/handlers/cache/CacheOperationPermissionRestCommandHandlerCheckTest.java

```diff
@@ -0,0 +1,442 @@
+
+import static java.util.Collections.singletonMap;
+import static org.apache.ignite.plugin.security.SecurityPermission.*;
+import static org.apache.ignite.testframework.GridTestUtils.assertThrowsWithCause;
+
+/**
+ * Test CRUD, create and destroy cache permissions with rest commands handler.
+ */
+public class CacheOperationPermissionRestCommandHandlerCheckTest extends GridCommonAbstractTest {
+    /** Empty permission. */
+    private static final SecurityPermission[] EMPTY_PERMS = new SecurityPermission[0];
+
+    /** Cache name for tests. */
+    private static final String CACHE_NAME = "TEST_CACHE";
+
+    /** Forbidden cache. */
+    private static final String FORBIDDEN_CACHE_NAME = "FORBIDDEN_TEST_CACHE";
+
+    /** New cache. */
+    private static final String NEW_TEST_CACHE = "NEW_TEST_CACHE";
+
+    /** Key. */
+    private String key = "key";
+
+    /** Value. */
+    private String val = "value";
+
+    /** New value. */
+    private String newVal = "newValue";
+
+    /** Cache permissions. */
+    private Map<String, SecurityPermission[]> cachePerms = new HashMap<>();
+
+    /** Security permission set. */
+    private Set<SecurityPermission> secPermSet = new HashSet<>();
+
+    /** Default allow all. */
+    private boolean dfltAllowAll;
+
+    /** Handler. */
+    private GridRestCommandHandler hnd;
+
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration() throws Exception {
+        TcpDiscoverySpi disco = new TcpDiscoverySpi();
+
+        disco.setIpFinder(new TcpDiscoveryVmIpFinder(true));
+
+        IgniteConfiguration cfg = super.getConfiguration();
+
+        cfg.setDiscoverySpi(disco);
+
+        SecurityPermissionSetBuilder builder = SecurityPermissionSetBuilder.create();
+
+        builder.defaultAllowAll(dfltAllowAll);
+
+        cachePerms.forEach((builder::appendCachePermissions));
+        secPermSet.forEach(builder::appendSystemPermissions);
+
+        cfg.setDataStorageConfiguration(
+            new DataStorageConfiguration()
+                .setDefaultDataRegionConfiguration(
+                    new DataRegionConfiguration()
+                        .setPersistenceEnabled(true)
+                )
+        )
+            .setAuthenticationEnabled(true)
+            .setPluginProviders(
+                new TestSecurityPluginProvider("login", "password", builder.build(), true));
+
+        return cfg;
+    }
+
+    /** */
+    @Test
+    public void testCacheCreate() throws Exception {
+        cachePerms.putIfAbsent(CACHE_NAME, new SecurityPermission[] {CACHE_CREATE});
+        cachePerms.putIfAbsent(FORBIDDEN_CACHE_NAME, EMPTY_PERMS);
+
+        startGrid(getConfiguration()).cluster().active(true);
+
+        assertTrue(grid().cacheNames().isEmpty());
+
+        cacheCreate(CACHE_NAME);
+
+        assertTrue(grid().cacheNames().contains(CACHE_NAME));
+
+        if (secPermSet.contains(CACHE_CREATE) || cachePermsContains(FORBIDDEN_CACHE_NAME, CACHE_CREATE)) {
```

Comment:
What this condition is for? The state of secPermSet and cachePerms is known at compile time. This relates to some tests below as well.